src/App.tsx

```diff
@@ -3,33 +3,50 @@ import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
-type State = {};
+type State = {
+  query: string;
+};
 
 export class App extends React.Component<{}, State> {
-  state: State = {};
+  state: State = {
+    query: '',
+  };
 
   render() {
+    const { query } = this.state;
+    const visibleMovies = moviesFromServer.filter(movie => (
+      movie.title.toLowerCase().includes(query.toLowerCase())
+      || movie.description.toLowerCase().includes(query.toLowerCase())));
```

Comment:
You can optimize this logic
```suggestion
    const visibleMovies = moviesFromServer.filter({ title, description } => {
      	const lowerCaseQuery = this.state.query.toLowerCase();
      	
        return (
          title.toLowerCase().includes(lowerCaseQuery)
          || description.toLowerCase().includes(lowerCaseQuery)
        );
      });
```

Reply:
Create a separate function `getVisibleMovies()` for this logic